dags/bqetl_artifact_deployment.py

```diff
@@ -56,11 +58,26 @@
 
 tags = [Tag.ImpactTier.tier_1]
 
+params = {
+    "generate_sql": Param(
+        default=False,
+        type="boolean",
+        description="Run sql generation before each publish task",
+    ),
+}
+
+# renders generate sql command if params.generate_sql is true, else empty string
+generate_sql_cmd_template = (
+    "{{ 'script/bqetl generate all --use-cloud-function=false && ' "
+    "if params.generate_sql else '' }}"
+)
+
 
-def check_for_queued_runs(dag_id: str) -> bool:
+def check_for_queued_runs(dag_id: str, generate_sql: bool) -> bool:
+    """Return true if there are no other queued dag runs or if the generate_sql param is true."""
     queued_runs = DagRun.find(dag_id=dag_id, state=DagRunState.QUEUED)
     print(f"Found {len(queued_runs)} queued dag runs for {dag_id}")
-    return len(queued_runs) == 0
+    return len(queued_runs) == 0 or generate_sql == "True"
```

Comment:
Is this a typo or is there some Airflow reason the `generate_sql` comparison is a `str` here but `bool` in the type hint? 
```suggestion
    return len(queued_runs) == 0 or generate_sql
```